engine/crates/engine-v2/src/engine/trusted_documents.rs

```diff
@@ -65,23 +65,47 @@ impl<'ctx, R: Runtime> PreExecutionContext<'ctx, R> {
                     Err(graphql_error)
                 }
             }
-            (true, Some(ext), _) => Ok(PreparedOperationDocument {
-                cache_key: Key::Operation {
-                    name,
-                    schema_version,
-                    document: Document::PersistedQueryExt(ext),
-                }
-                .to_string(),
-                document_fut: Some(self.handle_apollo_client_style_trusted_document_query(ext, client_name)?),
-            }),
-            (true, _, Some(document_id)) => Ok(PreparedOperationDocument {
+            // Apollo Client style trusted document query
+            (true, maybe_ext, maybe_doc_id) => {
+                let Some(client_name) = client_name else {
+                    return Err(GraphqlError::new(
+                        format!(
+                            "Trusted document queries must include the {} header",
+                            X_GRAFBASE_CLIENT_NAME.as_str()
+                        ),
+                        ErrorCode::TrustedDocumentError,
+                    ));
+                };
+
+                let doc_id = if let Some(ext) = maybe_ext {
```

Comment:
:bulb: **Performance**
The use of `Cow::Owned(hex::encode(&ext.sha256_hash))` in line 81 can be optimized. The `hex::encode` function creates a new `String` from the byte slice, which is then wrapped in a `Cow::Owned`. This involves unnecessary allocation. Instead, consider using a pre-allocated `String` and writing directly to it using `write!` macro, similar to the existing pattern in `handle_apollo_client_style_trusted_document_query`. This will reduce memory allocations and improve performance.
```suggestion
let mut doc_id = String::with_capacity(ext.sha256_hash.len() * 2);
for byte in &ext.sha256_hash {
    write!(doc_id, "{byte:02x}").expect("write to String to succeed");
}
let doc_id = Cow::Owned(doc_id);
```